### ISSUE TITLE
Support MySQL persistence for "tctl admin workflow" commands

### ISSUE THREAD

**Issue by schallert** [temporalio/temporal#1881]
**Is your feature request related to a problem? Please describe.**

We use MySQL as our persistence store for Temporal. We recently had an issue where we created workflows in a tight loop in a test environment. This led to hundreds of thousands of "bad" workflows being created. Other namespaces in the same cluster were affected, and workflows across all namespaces became unreliable.

Temporal server logs indicated persistence timeouts for various tasks. The MySQL slow query log showed queries taking > 10s for various queries on the `executions_visibility` table. Disk space increased by ~4x on the DB. Overall, not a good day for the database.

We attempted to use `tctl admin workflow delete` to start deleting bad workflows. Unfortunately, we quickly discovered that set of commands only supports Cassandra. We had been following https://github.com/temporalio/temporal/issues/598 from a previous similar issue but didn't realize it didn't apply to all commands.

Tihomir was very helpful over [Slack](https://temporalio.slack.com/archives/CTTJCPZQE/p1630503185004100?thread_ts=1630502520.003900&cid=CTTJCPZQE) and suggested we set namespace retention to 1d so that workflows are cleaned up sooner. Unfortunately, this meant our entire Temporal server is degraded for at least 24h until the workflows are cleaned up. While this is somewhat acceptable for our test environments, this would have caused a serious outage if it happened in production. This worries us.

**Describe the solution you'd like**

We would like at least one of the following options, in order of preference:

1. The ability to delete a set of workflows in a given namespace in a given time window from MySQL.
2. The ability to immediately delete a namespace and all associated workflow data from MySQL.
3. The `tctl admin wf delete` command to work with MySQL so that we can pass "bad" workflow IDs to it and try to clean up the DB.

**Describe alternatives you've considered**

We considered deleting all rows from `executions_visibility` where namespace ID was equal to the bad namespace in the bad time range. Unfortunately, the `history_node` table also seemed to grow significantly during the outage. Based on its schema we could not see how to delete rows related to the bad namespace. We also did not want to delete rows from `executions_visibility` and then risk Temporal not being able to GC the workflows the correct way if we did that.

**Additional context**

Happy to provide other details that would be helpful. I'm also available on the Temporal community slack.

**Comment by yiminc** [comment]
What do you mean by "bad workflows"? Why are they bad? Do they have corrupted state?
Could you terminate those workflows?
executions_visibility table is for visibility, that should not impact workflow executions. But if they share the same database, and it exhausted database resource, then you have trouble. Do you have way to temporarily stop calling visibility APIs? 
I don't think setting shorter retention here would help. The retention only apply to closed workflows and updated retention time only apply to newly closed workflows after the change.

**Comment by schallert** [comment]
@yiminc "Bad workflows" in this case meant a workflow that was unintentionally returning `ContinueAsNew` in a tight loop. Each workflow was empty, none of them were corrupted as far as I can tell, but there were hundreds of thousands of empty workflows.

Unfortunately it looks like retention didn't help here as you anticipated. We still saw workflows older than 24h. Eventually we decided to delete all rows from `executions_visibility` and immediately Temporal recovered. Below is a graph of p95 `persistence_latency_bucket` metrics as emitted by Temporal. As soon as we deleted the `executions_visibility` entries, we saw restored performance across the board:

![Screen Shot 2021-09-02 at 1 51 06 PM](https://user-images.githubusercontent.com/434973/131893154-029de6d7-e8e5-44ee-9065-2f52e2a518cd.png)

Now we're left with the problem of the `history_node` table being very large, and we're not sure if it will get cleaned up. We've patched the delete command and are passing the run IDs of the workflows we deleted into it: https://github.com/temporalio/temporal/commit/8268e028a8e1c062034927975917889a87a33be6. I'm not sure if this is doing anything, but it seems to be removing entries from the DB.

I say this all with the caveat that we're not sure if we're doing the right thing here, but our instance was in a degraded enough state that we had to take any actions that seemed reasonable. Thankfully the `executions_visibility` clearing I mentioned restored full service.

**Comment by schallert** [comment]
@yiminc "Bad workflows" in this case meant a workflow that was unintentionally returning `ContinueAsNew` in a tight loop. Each workflow was empty, none of them were corrupted as far as I can tell, but there were hundreds of thousands of empty workflows.

Unfortunately it looks like retention didn't help here as you anticipated. We still saw workflows older than 24h. Eventually we decided to delete all rows from `executions_visibility` and immediately Temporal recovered. Below is a graph of p95 `persistence_latency_bucket` metrics as emitted by Temporal. As soon as we deleted the `executions_visibility` entries, we saw restored performance across the board:

![131893154-029de6d7-e8e5-44ee-9065-2f52e2a518cd](https://user-images.githubusercontent.com/434973/131895333-b95bdf0c-f8f3-471e-ba6a-630376884057.png)


Now we're left with the problem of the `history_node` table being very large, and we're not sure if it will get cleaned up. We've patched the delete command and are passing the run IDs of the workflows we deleted into it: https://github.com/temporalio/temporal/commit/8268e028a8e1c062034927975917889a87a33be6. I'm not sure if this is doing anything, but it seems to be removing entries from the DB.

I say this all with the caveat that we're not sure if we're doing the right thing here, but our instance was in a degraded enough state that we had to take any actions that seemed reasonable. Thankfully the `executions_visibility` clearing I mentioned restored full service.
